supportedBackends/tes/src/main/scala/cromwell/backend/impl/tes/TesTask.scala

```diff
@@ -0,0 +1,225 @@
+        tesPaths.toContainerPath(f).valueString,
+        "File",
+        Some(false)
+      )
+    } ++ Seq(commandScript)
+
+  // TODO add TES logs to standard outputs
+  private val standardOutputs = Seq("rc", "stdout", "stderr").map {
+    f =>
+      TaskParameter(
+        f,
+        Some(fullyQualifiedTaskName + "." + f),
+        tesPaths.storageOutput(f),
+        tesPaths.containerOutput(f),
+        "File",
+        Some(false)
+      )
+  }
+
+  // extract output file variable names
+  private val outputFileNames = jobDescriptor.call.task.outputs
+      .filter {
+        case o => o.wdlType.toWdlString == "Array[File]" || o.wdlType.toWdlString == "File"
+      }
+    .map{case o => o.unqualifiedName}
+
+  // extract output files
+  private val outputWdlFiles = jobDescriptor.call.task
+    .findOutputFiles(jobDescriptor.fullyQualifiedInputs, NoFunctions)
+
+  private val wdlOutputs = outputFileNames.zip(outputWdlFiles)
+    .flatMap {
+      case (outputName, f: WdlSingleFile) => {
+        val outputFile = f.valueString
+        Seq(
+          TaskParameter(
+            outputName,
+            Some(fullyQualifiedTaskName + "." + outputName),
+            tesPaths.storageOutput(outputFile),
+            tesPaths.containerOutput(outputFile),
+            "File",
+            Some(false)
+          )
+        )
+      }
+      case (outputName, g: WdlGlobFile) => {
+        val globName = callEngineFunction.globName(g.value)
+        val globDirName = outputName + ".globDir"
+        val globDirectory = globName + "/"
+        val globListName = outputName + ".globList"
+        val globListFile = globName + ".list"
+        Seq(
+          TaskParameter(
+            globDirName,
+            Some(fullyQualifiedTaskName + "." + globDirName),
+            tesPaths.storageOutput(globDirectory),
+            tesPaths.containerOutput(globDirectory),
+            "Directory",
+            Some(false)
+          ),
+          TaskParameter(
+            globListName,
+            Some(fullyQualifiedTaskName + "." + globListName),
+            tesPaths.storageOutput(globListFile),
+            tesPaths.containerOutput(globListFile),
+            "File",
+            Some(false)
+          )
+        )
+      }
+    }
+
+  val outputs = wdlOutputs ++ standardOutputs
+
+  private val workingDirVolume = runtimeAttributes
+    .dockerWorkingDir
+    .map(path => Volume(
+      path,
+      // TODO all volumes currently get the same requirements
+      Some(runtimeAttributes.disk.to(MemoryUnit.GB).amount.toInt),
+      None,
+      path
+    ))
+
+  val volumes = Seq(
+    Volume(
+      tesPaths.dockerWorkflowRoot.toString,
+      Some(runtimeAttributes.disk.to(MemoryUnit.GB).amount.toInt),
+      None,
+      tesPaths.dockerWorkflowRoot.toString
+    )
+  ) ++ workingDirVolume
+
+  val resources = Resources(
+    runtimeAttributes.cpu,
+    runtimeAttributes.memory.to(MemoryUnit.GB).amount.toInt,
+    Some(false),
+    Some(volumes),
```

Comment:
I should also toss in why do I still say this for examples like `Some(false)` where it can't possibly be null. There it's more about building up the muscle memory and the uniformity of it. Admittedly I care less in those situations.